server/storage/sqldb.go

```diff
@@ -45,13 +45,44 @@ func translateOldVersionError(err error) error {
 	return err
 }
 
+// TufFilesInChannels scopes gorm queries to a particular channel or set of channels
+func TufFilesInChannels(channels ...*Channel) func(db *gorm.DB) *gorm.DB {
+	channelIDs := []uint{}
+	for _, channel := range channels {
+		channelIDs = append(channelIDs, channel.ID)
+	}
+
+	// If no channels were added, check Published
+	if len(channelIDs) == 0 {
+		channelIDs = []uint{Published.ID}
+	}
+
+	// If only one channel is provided, we don't need to group/collect
+	if len(channelIDs) == 1 {
+		return func(db *gorm.DB) *gorm.DB {
+			return db.Joins("INNER JOIN channels_tuf_files ON tuf_files.id = channels_tuf_files.tuf_file_id").
+				Where("channel_id = ?", channelIDs[0])
+		}
+	}
+
+	// Find tuf files matching all passed channels
+	return func(db *gorm.DB) *gorm.DB {
+		return db.Joins("INNER JOIN channels_tuf_files ON tuf_files.id = channels_tuf_files.tuf_file_id").
+			Where("channel_id IN (?)", channelIDs).
+			Group("tuf_files.id").Having("COUNT(DISTINCT channel_id) = ?", len(channelIDs))
+	}
+}
+
 // UpdateCurrent updates a single TUF.
 func (db *SQLStorage) UpdateCurrent(gun data.GUN, update MetaUpdate) error {
+	setDefaultChannels(&update)
 	// ensure we're not inserting an immediately old version - can't use the
 	// struct, because that only works with non-zero values, and Version
 	// can be 0.
-	exists := db.Where("gun = ? and role = ? and version >= ?",
-		gun.String(), update.Role.String(), update.Version).First(&TUFFile{})
+
+	exists := db.Scopes(TufFilesInChannels(update.Channels...)).
```

Comment:
I think these are very good points, and we should add additional checks for those cases when actually implementing the threshold support.